report_dynamic/models/report_dynamic_section.py

```diff
@@ -0,0 +1,217 @@
+import copy
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import UserError
+from odoo.tools import safe_eval
+
+from .header import Header
+
+try:
+    from jinja2.sandbox import SandboxedEnvironment
+
+    mako_template_env = SandboxedEnvironment(
+        block_start_string="<%",
+        block_end_string="%>",
+        variable_start_string="${",
+        variable_end_string="}",
+        comment_start_string="<%doc>",
+        comment_end_string="</%doc>",
+        line_statement_prefix="%",
+        line_comment_prefix="##",
+        trim_blocks=True,  # do not output newline after blocks
+        autoescape=True,  # XML/HTML automatic escaping
+    )
+    # Let's keep these in case they are needed
+    # in the future
+    mako_template_env.globals.update(
+        {
+            "str": str,
+            "len": len,
+            "abs": abs,
+            "min": min,
+            "max": max,
+            "sum": sum,
+            "filter": filter,
+            "map": map,
+            "round": round,
+            "page": "<p style='page-break-after:always;'/>",
+        }
+    )
+    mako_safe_template_env = copy.copy(mako_template_env)
+    mako_safe_template_env.autoescape = False
+except ImportError:
+    pass
+
+
+class ReportDynamicSection(models.Model):
+    _name = "report.dynamic.section"
+    _description = "Section blocks for report.dynamic"
+
+    _order = "sequence"
+
+    name = fields.Char("Name", required=True)
```

Comment:
I think we can remove this field to be required, sometimes the content of the block will be self explanatory